build/sbt-launch-lib.bash

```diff
@@ -111,11 +112,10 @@ addDebugger () {
 # a ham-fisted attempt to move some memory settings in concert
 # so they need not be dicked around with individually.
 get_mem_opts () {
-  local mem=${1:-2048}
-  local perm=$(( $mem / 4 ))
-  (( $perm > 256 )) || perm=256
-  (( $perm < 4096 )) || perm=4096
-  local codecache=$(( $perm / 2 ))
```

Comment:
Another advice is that you should not piggy-back orthogonal things into your PR.
I understand your intention, but you doesn't mention at all that you are trying to remove `local perm`.
Technically, what you needed was only two-line change `declare sbt_default_mem=2048` and `local mem=${1:-$sbt_default_mem}` in this file. 

In general, please start with the minimal PR and grow it when there is a reviewer' request.

I'd like to recommend to to minimize PR.